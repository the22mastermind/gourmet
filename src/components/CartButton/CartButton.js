import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import { FAB } from 'react-native-paper';
import PropTypes from 'prop-types';
import Cart from '../Cart/Cart';

const styles = StyleSheet.create({
  fab: {
    position: 'absolute',
    margin: 16,
    right: 0,
    bottom: 0,
    backgroundColor: '#a4c49a',
  },
});

const CartButton = ({ items }) => {
  const [visibleCart, setVisibleCart] = useState(false);
  
  return (
    visibleCart ? (
      <Cart setVisibleCart={setVisibleCart} />
    ) : (
      <FAB
        testID="cart-button"
        style={styles.fab}
        label={items ? JSON.stringify(items) : null}
        icon="shopping-outline"
        color="#f7f8f8"
        onPress={() => setVisibleCart(!visibleCart)}
      />
    )
  );
};

CartButton.propTypes = {
  items: PropTypes.number.isRequired,
};

export default CartButton;
