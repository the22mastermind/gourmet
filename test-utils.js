import React from 'react';
import { render } from '@testing-library/react-native';
import PaperProvider from 'react-native-paper/lib/commonjs/core/Provider';

const AllTheProviders = ({ children }) => (
  <PaperProvider>{children}</PaperProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
