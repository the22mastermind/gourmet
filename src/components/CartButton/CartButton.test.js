import React from 'react';
import { NavigationContainer} from '@react-navigation/native';
import { render, fireEvent, waitForElementToBeRemoved } from 'test-utils';
import CartButton from './CartButton';

// eslint-disable-next-line jest/no-disabled-tests
describe.skip('<CartButton /> component', () => {
  it('renders <CartButton /> component correctly', () => {
    const screen = render(<CartButton items={2} />);
  
    expect(screen.getByTestId('cart-button')).toBeDefined();
    expect(screen.queryByText('2')).toBeDefined();
  });
  
  it('should display cart contents correctly', async () => {
    const screen = render(<NavigationContainer><CartButton items={2} /></NavigationContainer>);
    
    fireEvent.press(screen.getByTestId('cart-button'));
  
    await waitForElementToBeRemoved(() => {
      expect(screen.getByTestId('cart-button')).toBeNull();
      expect(screen.queryByText('2')).toBeNull();
      expect(screen.queryByText('Your order')).toBeDefined();
      expect(screen.getByTestId('close-bottom-sheet')).toBeDefined();
    });
  });
  
  it('matches <CartButton /> component snapshot', async () => {
    const screen = render(<CartButton items={1} />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
