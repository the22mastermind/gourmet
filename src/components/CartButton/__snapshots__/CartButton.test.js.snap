// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartButton /> component matches <CartButton /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#a4c49a",
        "borderRadius": 28,
        "bottom": 0,
        "elevation": 6,
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessibilityLabel="1"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 28,
          },
        ]
      }
      testID="cart-button"
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 48,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#f7f8f8",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó±‡•
            </Text>
          </View>
        </View>
        <Text
          selectable={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "marginHorizontal": 8,
                },
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#f7f8f8",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
              ],
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
</View>
`;
